Packages/MIES/MIES_Utilities_File.ipf

```diff
@@ -252,27 +256,42 @@ Function/S GetAllFilesRecursivelyFromPath(string pathName, [string extension])
 
 	AssertOnAndClearRTError()
 	allFilesList = IndexedFile($pathName, -1, extension, "????", FILE_LIST_SEP); err = GetRTError(1)
-	WAVE/T allFilesInDir = ListToTextWave(allFilesList, FILE_LIST_SEP)
-	for(fileName : allFilesInDir)
-
-		fileOrPath = ResolveAlias(fileName, pathName = pathName)
-
-		if(isEmpty(fileOrPath))
-			// invalid shortcut, try next file
-			continue
-		endif
-
-		GetFileFolderInfo/P=$pathName/Q/Z fileOrPath
-		ASSERT(!V_Flag, "Error in GetFileFolderInfo")
 
-		if(V_isFile)
-			allFiles = AddListItem(S_path, allFiles, FILE_LIST_SEP, Inf)
-		elseif(V_isFolder)
-			foldersFromAlias = AddListItem(S_path, foldersFromAlias, FILE_LIST_SEP, Inf)
-		else
-			ASSERT(0, "Unexpected file type")
-		endif
-	endfor
+	if(isWindows && cmpstr(extension, "????") && cmpstr(extension, ".lnk"))
+		allFiles = AddPrefixToEachListItem(S_path, allFilesList, sep = FILE_LIST_SEP)
+	else
+		// try to resolve aliases/shortcuts when we can have them
+		WAVE/T allFilesInDir = ListToTextWave(allFilesList, FILE_LIST_SEP)
+		for(fileName : allFilesInDir)
+
+			GetFileFolderInfo/P=$pathName/Q/Z fileName
+
+			if(V_Flag != 0)
+				// error querying file
+				continue
+			endif
+
+			if(V_IsAliasShortcut)
+				fileOrPath = ResolveAlias(fileName, pathName = pathName)
```

Comment:
There should be an early return here if ResolveALias returns ""

Reply:
Fixed.